hack/update-vendor.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+###
+#Copyright 2019 The KubeEdge Authors.
+#
+#Licensed under the Apache License, Version 2.0 (the "License");
+#you may not use this file except in compliance with the License.
+#You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#Unless required by applicable law or agreed to in writing, software
+#distributed under the License is distributed on an "AS IS" BASIS,
+#WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#See the License for the specific language governing permissions and
+#limitations under the License.
+###
+
+# The root of the build/dist directory
+KUBEEDGE_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd -P)"
+
+#KUBEEDGE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+
+# list_staging_repos outputs a sorted list of repos in staging/src/kubeedge
+# each entry will just be the $repo portion of staging/src/kubeedge/$repo/...
+# $KUBEEDGE_ROOT must be set.
+function kubeedge::util::list_staging_repos() {
+  (
+    cd "${KUBEEDGE_ROOT}/staging/src/github.com/kubeedge/" && \
+    find . -mindepth 1 -maxdepth 1 -type d | cut -c 3- | sort
```

Comment:
you can merge these two lines by editing the find command to this:
`find ${KUBEEDGE_ROOT}/staging/src/github.com/kubeedge/ -mindepth 1 -maxdepth 1 -type d | cut -c 3- | sort`

Reply:
I didn't like to change the directory here beacuse all commands which are executed after this has an different working directory and it could easily overlooked.

Reply:
@subpathdev  it is execd in (), so did not chang directory

Reply:
I will ask someone, who has a greater experience in bash, but if this did not change the directory you are still on on root directory of the repository and command on line 57 will not work.

Reply:
@subpathdev i can not find line 57 .

Reply:
@subpathdev i can not find line 57

Reply:
I mean 47 sry

Reply:
is resolved; The ln command is crasy;
It is possible to leave the cd command away and using this:
`find ${KUBEEDGE_ROOT}/staging/src/github.com/kubeedge/ -mindepth 1 -maxdepth 1 -type d -printf '%P\n' | cut -c 3- | sort`